src/cdk/v2/destinations/algolia/rtWorkflow.yaml

```diff
@@ -28,10 +27,47 @@ steps:
       $.outputs.transform#idx.error.(
         $.handleRtTfSingleEventError(^[idx], .originalError ?? ., {})
       )[]
-  - name: batchSuccessfulEvents
+
+  - name: separateDontBatchEventForProxy
```

Comment:
```
bindings:
  - path: ../../../../v0/destinations/algolia/config
  - name: handleRtTfSingleEventError
    path: ../../../../v0/util/index
steps:
  - name: validateInput
    template: |
      $.assert(Array.isArray(^) && ^.length > 0, "Invalid event array")

  - name: transform
    externalWorkflow:
      path: ./procWorkflow.yaml
      bindings:
        - name: batchMode
          value: true
    loopOverInput: true

  - name: successfulEvents
    template: |
      $.outputs.transform#idx.output.({
        "output": .,
        "destination": ^[idx].destination,
        "metadata": ^[idx].metadata
      })[]
  - name: failedEvents
    template: |
      $.outputs.transform#idx.error.(
        $.handleRtTfSingleEventError(^[idx], .originalError ?? ., {})
      )[]

  - name: batchSuccessfulEvents
    description: Batches the successfulEvents
    template: |
      const dontBatchTrueEvents =  $.outputs.successfulEvents{.metadata.dontBatch}[];
      const dontBatchFalseEvents = $.outputs.successfulEvents{!.metadata.dontBatch}[];
      
      let batches = [...$.chunk(dontBatchFalseEvents, $.MAX_BATCH_SIZE), ...dontBatchTrueEvents];
      batches@batch.({
        "batchedRequest": {
          "body": {
            "JSON": {"events": ~r batch.output[]},
            "JSON_ARRAY": {},
            "XML": {},
            "FORM": {}
          },
          "version": "1",
          "type": "REST",
          "method": "POST",
          "endpoint": {{$.ENDPOINT}},
          "headers": batch[0].destination.Config.().({
            "X-Algolia-Application-Id": .applicationId,
            "X-Algolia-API-Key": .apiKey
          }),
          "params": {},
          "files": {}
        },
        "metadata": ~r batch.metadata[],
        "batched": true,
        "statusCode": 200,
        "destination": batch[0].destination
      })[];

  - name: finalPayload
    template: |
      [...$.outputs.failedEvents, ...$.outputs.batchSuccessfulEvents]
```

try this, there are few tests failing but you will get idea

Reply:
done